### ISSUE TITLE
 memcg_stress_test : Cannot allocate memory

### ISSUE THREAD

**Issue by ouyangkui** [linux-test-project/ltp#688]
hi
When I run this case, failed with following error message 

memcg_stress_test 1 TINFO: timeout per run is 0h 35m 0s
memcg_stress_test 1 TINFO: Calculated available memory 514175 MB
memcg_stress_test 1 TINFO: Testing 150 cgroups, using 3426 MB, interval 5
memcg_stress_test 1 TINFO: Starting cgroups
memcg_stress_test 1 TINFO: Testing cgroups for 900s
memcg_stress_test.sh: line 65: 45033 User defined signal 1   memcg_process_stress $mem_size $interval

memcg_stress_test 1 TINFO: Killing groups
memcg_stress_test.sh: fork: Cannot allocate memory
memcg_stress_test 1 TBROK: test interrupted
memcg_stress_test.sh: fork: Cannot allocate memory

Actually in my system,  available memory less than free memory
free -h
             total        used        free      shared  buff/cache   available
Mem:           503G        745M        502G        2.5M        461M        499G
Swap:            0B          0B          0B

And I found info from https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=34e431b0ae398fc54ea69ff85ec700722c9da773 
available memory not equal to free + cache

So in new kernel,    >3.1.4   
use MEM =  min[ MemAvailable,  MemFree+SwapFree/2 ]   for testing , is it ok？

**Comment by manjuhr1** [comment]
Similar issue is seen in IBM ppc64le test servers, particularly when this test is executed in parallel with any other LTP tests:  It leads to Out Of Memory situation and eventually OOM kill will kill the test and other processes.

Currently testcase is getting usage memory (from /proc/meminfo) for testing as:  MEM=$(( $mem_free + $swap_free / 2 ))
But if we consider to use MEM = min[ MemAvailable, MemFree+SwapFree/2 ] ,   there might be another challenge:
Example, on my test VM, where total memory is 17G, but the 'MemAvailable' in /proc/meminfo is 549764202368 kB
# cat /proc/meminfo |grep MemAvailable
MemAvailable:   549764202368 kB

root@vimlp1:/SLS_PYTHON free -h
              total        used        free      shared  buff/cache   available
Mem:           17Gi       9.4Gi       8.0Gi        11Mi       512Ti       8.0Gi
Swap:         9.8Gi       619Mi       9.2Gi

For some reason cache memory and MemAvailable in /proc/meminfo has bounced up to around 512TB (Probably a bug? I am trying to recreate this issue). Please check if this case occurs in your VMs too.. If this is a frequent case testcase might end up picking : MemFree+SwapFree/2 

But at least from my test experience, running memcg_stress_test along with other tests will easily lead to OOM.

Thanks,
Manju

**Comment by ouyangkui** [comment]
Thanks
In my VM ,  no issue that you mentioned. 
AvailableMemroy is same in cmd free - h and /proc/meminfo . 

BTW, I found another issue for cleanup memcg folder. 

for i in $(seq 0 $(($cgroups-1))); do
		mkdir /dev/memcg/$i 2> /dev/null
		memcg_process_stress $mem_size $interval &
		echo $! > /dev/memcg/$i/tasks
		pids="$! $pids"
	done

when create 150 folder, pids list secquence is different with memcg numbers. 
In cleanup,  
When try to delete group 1,  will failed , because the 150th process will be killed  and the first process is running still.

modify pids="$! $pids" to pids="$pids $!" , will fix this issue.
